### DISCUSSION TITLE
Remapping i2s pins

### DISCUSSION THREAD

**Discussion by MichaelPNolan** [pschatzmann/arduino-audio-tools#249] [Q&A] [answered]
I have built other audio projects where I used everything up - for example I had a screen, USB Host and midi.
The screen was I2C so pin 22 is used 
I tried to remap to the same pins I normally used for the i2s dac on other projects but couldn't get any sound.
So eventually I relented and used what was defined when #define ESP32

AudioTools ESP32?? 
Bclk:14
WS:15
Din:22  (which is actually connected to your PIN_I2S_DATA_OUT - makes sense but nobody else did that in other examples).

Which made the audio sinewave with streams-generator-i2s work when I wired it up that way.
I'd like to go back to using my usual pins for i2s - can anyone explain why over-riding these won't work, should I fork and modify an AudioConfig.h?
//#define PIN_I2S_BCK 25
//#define PIN_I2S_WS 27
//#define PIN_I2S_DATA_OUT 26

I'm looking forward to exploring these tools because I want to do my own sine wave math library and learn, rather than pull in code from others which I can't seem to really follow because it's obscure to me. I'd like to make ADSL drum like sounds for something I built. I tried with an RPi Pico and circuit python but when it came to i2s the results were underwhelming so I'm back to restart it on the good ole ESP32 that i used for many other projects successfully.

Thanks for any suggestions. I did look through the discussions but didn't find so sorry if this is redundant
Regards
MichaelPNolan (that's my GitHub ID)

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you read the documentation:
https://github.com/pschatzmann/arduino-audio-tools/wiki/External-DAC

**Comment by MichaelPNolan** [comment]
I guess i made a mistake when I tried this:

  ```
auto config = i2sStream.defaultConfig(TX_MODE);
  config.pin_bck = 14;
  config.pin_ws = 15;
  config.pin_data = 22;
  i2sStream.begin(config);
```

I'll try it some more. Thanks
